File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        n = len(s)
        min_adds_required = 0
        count_open = 0
        
        for i in range(n):
            if s[i] == '(':
                count_open += 1
            else:
                if count_open > 0:
                    count_open -= 1
                else:
                    min_adds_required += 1
        
        return min_adds_required + count_open

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.c
#include <string.h>

int minAddToMakeValid(char* s) {
    int openBrackets = 0;
        int minAddsRequired = 0;
        int size = strlen(s);
        for(int i = 0; i<size; i++){
            if(s[i] == '('){
                openBrackets++;
            }else{
                openBrackets > 0 ? openBrackets-- : minAddsRequired++;
            }
        }
        return openBrackets + minAddsRequired;
}